main.go

```diff
@@ -140,6 +145,36 @@ func init() {
 		zlog.Fatalf("--connections-per-host must be in the range [0,50]")
 	}
 
+	// If a source IP is specified, verify that the host is advertising the IP
+	// and has a chance to receive the responses.
+	if sourceIP != "" {
+		config.LocalAddressSet = false
+		providedAddr := net.ParseIP(sourceIP)
+		localInterfaces, localInterfacesErr := net.Interfaces()
+		if localInterfacesErr != nil {
+			zlog.Fatalf("Error getting Local interfaces: %s", localInterfacesErr.Error())
+		}
+		for idx, i := range localInterfaces {
+			addresses, adddressesErr := i.Addrs()
+			if adddressesErr != nil {
+				zlog.Fatalf("Error getting Local addresses for interface %d: %s", idx, adddressesErr.Error())
+			}
+			for _, address := range addresses {
+				castAddress := address.(*net.IPNet)
+				if castAddress.Contains(providedAddr) {
+					tcpAddr := &net.TCPAddr{
+						IP: address.(*net.IPNet).IP,
```

Comment:
This appears to work. I thought I tried that and got an odd error. I will roll that into the new version.